worlds/tloz/Rules.py

```diff
@@ -0,0 +1,145 @@
+from typing import TYPE_CHECKING
+
+from ..generic.Rules import add_rule
+from .Locations import food_locations, shop_locations
+from .ItemPool import dangerous_weapon_locations
+
+if TYPE_CHECKING:
+    from . import TLoZWorld
+
+def set_rules(tloz_world: "TLoZWorld"):
+    player = tloz_world.player
+    world = tloz_world.multiworld
+
+    # Boss events for a nicer spoiler log play through
+    for level in range(1, 9):
+        boss = world.get_location(f"Level {level} Boss", player)
+        boss_event = world.get_location(f"Level {level} Boss Status", player)
+        status = tloz_world.create_event(f"Boss {level} Defeated")
+        boss_event.place_locked_item(status)
+        add_rule(boss_event, lambda state: state.can_reach(boss, "Location", player))
+
+    # No dungeons without weapons except for the dangerous weapon locations if we're dangerous, no unsafe dungeons
+    for i, level in enumerate(tloz_world.levels[1:10]):
+        for location in level.locations:
+            if world.StartingPosition[player] < 1 or location.name not in dangerous_weapon_locations:
+                add_rule(world.get_location(location.name, player),
+                         lambda state: state.has_group("weapons", player))
+            if i > 0:  # Don't need an extra heart for Level 1
+                add_rule(world.get_location(location.name, player),
+                         lambda state: state.has("Heart Container", player, i) or
```

Comment:
yet again using i from a `for` loop unsafely, going to be 9 or whatever for all of them